### DISCUSSION TITLE
How to use an endpoint with url parameters in Hasura actions

### DISCUSSION THREAD

**Discussion by Tristangva** [hasura/graphql-engine#9346] [Question] [answered]
I am at the moment trying to convert an endpoint like https://example.com/check?param1=value1 to graphql with Actions. To fill value1, I want to use an input from an input type as I configured in the type configuration. I am unable to find any examples online using this format.

**Comment by meetzaveri** [comment, marked as answer]
Hey @Tristangva 

If I am correct, this - https://example.com/check would be your webhook/action handler for your action right ?

### Request URL transform
Acc. to your use case, you can use "Request URL" transform in your action. Reference - https://hasura.io/docs/latest/actions/rest-connectors/#request-url

So if you want to have `param1` to have value supplied from your request's body input  say `arg1`, you can supply query param under "Change Request Options" section while creating action via console.

See attached screenshot

![captures_chrome-capture-2023-0-2](https://user-images.githubusercontent.com/21293048/210235689-30861175-205b-43bb-af54-b5d33f2574de.png)

So your GraphQL mutation would look like
```gql
mutation MyMutation {
  testAction(arg1: {value: "value123"}) {
    name
  }
}
```
where testAction is my action. And upon hitting your endpoint, in your backend, it will have `req.query` pointing to `{ param1: 'value123' }`. In your backend, assuming that `req` is reference to request param passed into function as argument (like `function testActionHandler(req, res)`)

Is this what you were trying to achieve ? Like Passing value, supplied from gql mutation to your custom param ?

----
Or I might be wrong in understanding your use-case, so do let us know, thanks !

**Reply by Tristangva** [reply]
This is great! Thanks!
